packages/data-stores/src/domain-suggestions/resolvers.ts

```diff
@@ -16,13 +17,23 @@ import {
 } from './actions';
 import { fetchAndParse, wpcomRequest } from '../wpcom-request-controls';
 import { getFormattedPrice } from './utils';
-
 import type { DomainSuggestion, DomainSuggestionQuery } from './types';
 
-export const isAvailable = function* isAvailable( domainName: string ) {
-	const url = `https://public-api.wordpress.com/rest/v1.3/domains/${ encodeURIComponent(
+function getAvailabilityURL( domainName: string ) {
+	return `https://public-api.wordpress.com/rest/v1.3/domains/${ encodeURIComponent(
 		domainName
 	) }/is-available?is_cart_pre_check=true`;
+}
+
+function suggestionsLackThisFQDN( suggestions: DomainSuggestion[], domainName: string ) {
+	return (
+		validator.isFQDN( domainName ) &&
+		! suggestions.some( ( s ) => s.domain_name.toLowerCase() === domainName )
+	);
+}
+
+export const isAvailable = function* isAvailable( domainName: string ) {
```

Comment:
This is called when the `isAvailable` selector in `packages/data-stores/src/domain-suggestions/selectors.ts` is called.

Reply:
Yes